pkg/thread/threadmodel.go

```diff
@@ -48,7 +48,7 @@ func GetModelAndModelProviderForThread(ctx context.Context, c kclient.Client, th
 	model := thread.Spec.Manifest.Model
 
 	// If it wasn't set on the thread, try to find a parent project that has it set.
-	if modelProvider == "" || model == "" {
```

Comment:
@thedadams since we are not setting model provider, we should prevent it from falling back to default model when it is empty. I missed that when addressing your comment last time.